packages/block-library/src/query/edit.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * Internal dependencies
+ */
+import QueryPanel from './query-panel';
+
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { debounce, isUndefined, pickBy } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	BlockList,
+	BlockEditorProvider,
+	InspectorControls,
+	WritingFlow,
+} from '@wordpress/block-editor';
+import { cloneBlock, createBlock } from '@wordpress/blocks';
+import { PanelBody, Placeholder, Spinner } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { EntityProvider } from '@wordpress/core-data';
+import { withSelect } from '@wordpress/data';
+
+const defaultFields = [
+	'core/post-title',
+	'core/post-date',
+	'core/post-author',
+	'core/post-excerpt',
+];
+
+class Edit extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			editingPost: null,
+			blocksTree: {},
+		};
+
+		this.debouncedCreateBlockTree = debounce(
+			this.createBlockTree.bind( this ),
+			1000
+		);
+	}
+
+	componentDidMount() {
+		this.createBlockTree();
+		this.updateBlocks( defaultFields.map( ( f ) => createBlock( f ) ) );
```

Comment:
Instead of doing this onMount, maybe this could be the default valaue of the attribute.